File: app/src/main/java/mobsoft/bme/hu/irodafoglalo/interactor/InteractorModule.java
package mobsoft.bme.hu.irodafoglalo.interactor;

/**
 * Created by mobsoft on 2016. 04. 18..
 */

import dagger.Module;
import dagger.Provides;
import mobsoft.bme.hu.irodafoglalo.interactor.rooms.RoomsInteractor;

@Module
public class InteractorModule {

    @Provides
    public RoomsInteractor provideRoomInteractor() {
        return new RoomsInteractor();
    }
}

File: app/src/main/kotlin/mobsoft/bme/hu/irodafoglalo/interactor/InteractorModule.kt
package mobsoft.bme.hu.irodafoglalo.interactor

import dagger.Module
import dagger.Provides
import mobsoft.bme.hu.irodafoglalo.interactor.db.DBInteractor
import mobsoft.bme.hu.irodafoglalo.interactor.rooms.RoomsInteractor

/**
 * Created by Adam Toth on 2016. 04. 19..
 */
@Module
class InteractorModule {

    @Provides
    fun provideRoomsInteractor(): RoomsInteractor {
        return RoomsInteractor()
    }

    @Provides
    fun provideDBInteractor(): DBInteractor {
        return DBInteractor()
    }
}
